### DISCUSSION TITLE
Parse from json

### DISCUSSION THREAD

**Discussion by Kolhun** [nlohmann/json#4252] [Q&A]
Hello everyone, a question arose when analyzing json, specifically when downloading from the network it gives an error that it is not possible to read a forbidden character, can these be characters like quotation marks or does it indicate that there are no quotes at all? I will indicate the error and the output below. How can this be fixed? Or am I doing something wrong?

Console output:

curl_easy_perform() failed:Timeout was reached
curl exception [json.exception.parse_error.101] parse error at line 1, column 5613: syntax error while parsing value - invalid string: missing closing quote; last read: '"\u041e\u043f\u0438\u0441\u0430\u043d\u0438\u0435<br>\u0410\u0440\u0431\u0430\u043b\u0435\u0442 \u043f\u0438\u0441\u0442\u043e\u043b\u0435\u0442\u043d\u043e\u0439 \u043a\u043e\u043d\u0441\u0442\u0440\u0443\u043a\u0446\u0438\u0438 \u00ab\u0410\u0441\u043f\u0438\u0434\u00bb \u0438\u043c\u0435\u0435\u0442 \u0432\u043e\u0437\u043c\u043e\u0436\u043d\u043e\u0441\u0442\u044c \u0441\u0442\u0440\u0435\u043b\u044c\u0431\u044b \u0434\u0440\u043e\u0442\u0438\u043a\u0430\u043c\u0438, \u0448\u0430\u0440\u0438\u043a\u0430\u043c\u0438 \u0438 \u0433\u0430\u0440\u043f\u0443\u043d\u0430\u043c\u0438. \u0413\u0430\u0440\u043f\u0443\u043d\u044b \u0438\u0441\u043f\u043e\u043b\u044c\u0437\u0443\u044e\u0442\u0441\u044f \u0434\u043b\u044f \u043b\u043e\u0432\u043b\u0438 \u0440\u044b\u0431\u044b, \u043f\u0440\u0438 \u044d\u0442\u043e\u043c \u0430\u0440\u0431\u0430\u043b\u0435\u0442 \u00ab\u0410\u0441\u043f\u0438\u0434\u00bb \u043e\u0441\u043d\u0430\u0449\u0430\u0435\u0442\u0441\u044f \u0431\u0435\u0437\u0438\u043d\u0435\u0440\u0446\u0438\u043e\u043d\u043d\u043e\u0439 \u043a\u0430\u0442\u0443\u0448\u043a\u043e\u0439. \u0414\u0430\u043b\u044c\u043d\u043e\u0441\u0442\u044c \u043f\u0440\u0438\u0446\u0435\u043b\u044c\u043d\u043e\u0439 \u0441\u0442\u0440\u0435\u043b\u044c\u0431\u044b \u0430\u0440\u0431\u0430\u043b\u0435\u0442 \u0430\u0441\u043f\u0438\u0434 \u0441\u043e\u0441\u0442\u0430\u0432\u043b\u044f\u0435\u0442 \u043f\u043e\u043b\u0442\u043e\u0440\u0430 \u0434\u0435\u0441\u044f\u0442\u043a\u0430 \u043c\u0435\u0442\u0440\u043e\u0432, \u0430\u0440\u0431\u0430\u043b\u0435\u0442 \u00ab\u0410\u0441\u043f\u0438\u0434\u00bb \u043e\u0442\u043b\u0438\u0447\u0430\u0435\u0442\u0441\u044f \u0432\u044b\u0441\u043e\u043a\u043e\u0439 \u0442\u043e\u0447\u043d\u043e\u0441\u0442\u044c\u044e \u0438 '.

Output data console from unicode in UTF-8:
![image](https://github.com/nlohmann/json/assets/36815965/63888d45-be9d-449d-9c77-f65ebd7feb86)
![image](https://github.com/nlohmann/json/assets/36815965/31ec88b0-c53b-4485-976c-175f1ae90027)

**Comment by nlohmann** [comment]
The log line above the parse error already shows the cause: 'curl_easy_perform() failed: Timeout was reached'. The HTTP request was cut off mid-response, so the buffer handed to json::parse contains a truncated JSON string (the closing quote for that long Cyrillic string literal never arrived), which is exactly what 'missing closing quote' at column 5613 means. This isn't a bug in the JSON parser or an escaping issue — you need to fix the curl timeout/retry logic (increase CURLOPT_TIMEOUT or handle partial downloads) so the full response body is received before it's parsed.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
